### DISCUSSION TITLE
How to make the timeline accurate to milliseconds

### DISCUSSION THREAD

**Discussion by djkcyl** [openai/whisper#321] [Q&A]
As the question shows, I need to output a more precise time
![1%J)1PIFTI_TB}W2T(SFJVF](https://user-images.githubusercontent.com/59153990/195862983-9df86fe9-fee0-4dc1-be14-3425d1659317.png)

**Comment by NielsMayer** [comment]
Although it won't help with accuracy, I wanted the time positions in milliseconds, instead of the slowness of parsing and storage size of having times in human readable format, when internally, I need timings in milliseconds to begin with, for #233 .

See my pull request at #228

**Comment by davtoro** [comment]
I've the same problem, all timestamps have 000 milliseconds and they cut words because of that or they start at .000 and have additional word sometimes because they should start at .420 etc.

Is it technically possible to make it more accurate just changing the code? or this is some property of the model and it will not be more accurate?

**Comment by Baenwort** [comment]
Take a look at https://github.com/openai/whisper/discussions/435 it might be an effort you can get some ideas or switch to using as it seems focused on timestamp accuracy

**Comment by davtoro** [comment]
Thanks, I've seen that and tried it but it's even less accurate than default whisper. The problem is that I'm cutting audio based on the timestamps as it is used as training data for another model, so I could get 200 audio files from a 10 minute audio, that's why milliseconds precision is so important as cutting at .000 cuts words from the audio while they still exist in text.

Problem is that transcribe function results have only seconds precision in start/end variables e.g. `'id': 132, 'seek': 66300, 'start': 679.0, 'end': 682.0, `, number of milliseconds is always zero.

**Comment by nagisa-eevee** [comment]
If you are using whisper in command-line, try to add `--word_timestamps True`. It would be much better, but as mentioned in #314, whisper timestamp for the end of the dialogue is usually quite accurate, while it usually gets the start wrong.
